### PR TITLE
Fix a bug in the marsh_flooding/idealized_transect case

### PR DESCRIPTION
**Pull request by caozd999**
This bug appeared in linking the python script **`comparison.py`** to the **`script_test_dir`** directory. It intended to provide the path of the python script by using **`path_base`**, which generated to a invalid hyperlink of the script and failed to run the analysis. Changing it to **`source_path`** fixed this issue.

### CODE REVIEW

**Comment by xylar** [comment]
@caozd999, can you comment here on what testing you did to ensure that this works as expected now?

**Comment by xylar** [comment]
## Testing

I tested on my linux laptop using gnu compilers.  I set up the test cases as:
```
$ ./list_testcases.py | grep idealized_transect
  68: -o ocean -c drying_slope -r marsh_flooding -t idealized_transect
  71: -o ocean -c drying_slope -r meshes -t idealized_transect
```
```
$ ./setup_testcase.py -f config.ocean -n 68,71 -m runtime_definitions/mpirun.xml --work_dir ~/data/mpas/test_marsh
 -- Set up case: /home/xylar/data/mpas/test_marsh/ocean/drying_slope/marsh_flooding/idealized_transect/forward2
 -- Set up case: /home/xylar/data/mpas/test_marsh/ocean/drying_slope/marsh_flooding/idealized_transect/analysis
 -- Set up case: /home/xylar/data/mpas/test_marsh/ocean/drying_slope/marsh_flooding/idealized_transect/init_step1
 -- Set up case: /home/xylar/data/mpas/test_marsh/ocean/drying_slope/marsh_flooding/idealized_transect/forward1
 -- Set up case: /home/xylar/data/mpas/test_marsh/ocean/drying_slope/marsh_flooding/idealized_transect/init_step3
 -- Set up case: /home/xylar/data/mpas/test_marsh/ocean/drying_slope/marsh_flooding/idealized_transect/forward3
 -- Set up case: /home/xylar/data/mpas/test_marsh/ocean/drying_slope/marsh_flooding/idealized_transect/init_step2
 -- Set up driver script in /home/xylar/data/mpas/test_marsh/ocean/drying_slope/marsh_flooding/idealized_transect
 -- Set up case: /home/xylar/data/mpas/test_marsh/ocean/drying_slope/meshes/idealized_transect/build_transect_mesh
 -- Set up driver script in /home/xylar/data/mpas/test_marsh/ocean/drying_slope/meshes/idealized_transect
```
I had to reduce the number of MPI tasks from 36 to 4 to run well on my laptop.  Given the size of the mesh (`nCells = 464`), I think even 4 processors is probably too many to be more efficient than 2 or even 1.  @caozd999, I would recommend reducing the number of MPI tasks to 2.

I can verify that the script `comparison.py` in the `analysis` step is linked correctly.

The tests are still running but I will update here when they complete successfully.

**Comment by xylar** [comment]
The "failed" Azure test seems to be a GitHub reporting glitch.  The test passed just fine when I actually look at the run on Azure.

**Review by xylar — APPROVED** [review]
The fix works.  I'll merge but let me know first if you want to reduce the number of MPI tasks here before I merge.

**Comment by caozd999** [comment]
@xylar Thanks for testing this branch. I changed the MPI to 2 and update the PR.

**Comment by xylar** [comment]
@caozd999, could you retest the whole test case with this latest change?  If it works for you, I'll merge tomorrow.  If you have trouble with the new compass repo, let me know and I'll try to help you out.  If you can't figure it out before tomorrow morning, I'll go ahead and retest tomorrow and merge if it works for me.

**Comment by xylar** [comment]
@caozd999, I reran the test cases with the new MPI task count and everything ran fine for me.  This time, I ran the 3 `forward*` steps at the same time, each on 2 cores.  It was a little slower per run than running one at a time on 4 cores but a little faster for the 3 combined (1 hour 50 min).  But I suspect that's because my laptop actually only has 4 cores so running in 6 cores isn't really a great idea.  Anyway, everything looks good and I'll merge.

**Comment by xylar** [comment]
You should probably delete your branch to keep your fork tidy.

**Pull request merged**
